src/__tests__/showInApp.test.ts

```diff
@@ -185,6 +186,114 @@ describe('defaults', () => {
   });
 });
 
+describe('Ask for web push notification permission', () => {
```

Comment:
This test's description does not match what it does or does not describe what it expects.

Reply:
The below two tests look great, but this one is very confusing in general. I suggest breaking it down to the various user scenarios. Example:

- When askForWebPushPermission and localStorage is not set, opt-in message is displayed
- When askForWebPushPermission and localStorage is set, opt-in message is not displayed
- When opt-in YES is clicked, ...
- When opt-in NO is clicked, ...

Avoid logic and iteration in tests.